content/articles/session-management-in-nodejs-using-expressjs-and-express-session/index.md

```diff
@@ -0,0 +1,321 @@
+
+```js
+// creating 24 hours from milliseconds
+const oneDay = 1000 * 60 * 60 * 24;
+```
+
+Check the [documentation](https://www.npmjs.com/package/express-session#options) for all possible options and learn more about these options.
+
+### Setting up the session middleware
+
+To initialize the session, we will set the session middleware inside the routes of the individual HTTP requests.
+
+When a client sends a request, the server will set a session ID and set the cookie equal to that session ID. The cookie is then stored in the set cookie HTTP header in the browser. Every time the browser (client) refreshes, the stored cookie will be a part of that request.
+
+We'll create a simple login form to demonstrate that.
+
+Here is the login form (`index.html`).
+
+```html
+<html>
+<head>
+    <link rel="stylesheet" href="app.css">
+</head>
+<body>
+    <form action="/user" method="post">
+        <h2>Login</h2>
+        <div class="input-field">
+            <input type="text" name="username" id="username" placeholder="Enter Username">
+        </div>
+        <div class="input-field">
+            <input type="password" name="password" id="password" placeholder="Enter Password">
+        </div>
+        <input type="submit" value="LogIn">
+    </form>
+</body>
+</html>
+```
+
+And some CSS to style the form (`app.css`).
+
+```css
+body {
+    display: flex;
+    justify-content: center;
+}
+
+form {
+    display: flex;
+    flex-direction: column;
+}
+
+.input-field {
+    position: relative;
+    margin-top: 2rem;
+}
+
+.input-field input {
+    padding: 0.8rem;
+}
+
+form .input-field:first-child {
+    margin-bottom: 1.5rem;
+}
+
+form input[type="submit"] {
+    background: linear-gradient(to left, #4776E6, #8e54e9);
+    color: white;
+    border-radius: 4px;
+    margin-top: 2rem;
+    padding: 0.4rem;
+}
+```
+
+Let's setup the server. Create an `app.js` file and set up the session server, as shown below.
+
+#### Import all the Node.js libraries that we explained earlier
+
+```js
+const express = require('express');
+const cookieParser = require("cookie-parser");
+const bodyParser=require('body-parser');
+const sessions = require('express-session');
+```
+
+#### Initialize the express app
+
+```js
+const app = express();
+```
+
+#### Add the Express-session options
+
+```js
+// creating 24 hours from milliseconds
+const oneDay = 1000 * 60 * 60 * 24;
+
+//session middleware
+app.use(sessions({
+    secret: "thisismysecrctekeyfhrgfgrfrty84fwir767",
+    saveUninitialized:true,
+    cookie: { maxAge: oneDay },
+    resave: false
+}));
+```
+
+#### Parse the HTML form
+As we explained, the Body-parser will help us parser an HTTP POST method request from an HTML document. Set the sever to use the Body-parser so that we can access this POST method. We also need to serve the CSS styling to format the outlook of the HTML form.
+
+```js
+// get our app to use a body-parser
+
+app.use(bodyParser.urlencoded({ extended: true }))
```

Comment:
BodyParser is deprecated. Please use express methods to perform this operation. 
Remove it from the project dependencies throughout the article.